src/useCookie.ts

```diff
@@ -0,0 +1,125 @@
+import { useState, useEffect } from 'react'
+
+import noop from './shared/noop'
+import isClient from './shared/isClient'
+import isDevelopment from './shared/isDevelopment'
+import isAPISupported from './shared/isAPISupported'
+import createHandlerSetter from './factory/createHandlerSetter'
+
+import { BRHGeolocationPositionError } from './shared/types'
+
+export enum ECookieSameSite {
+  STRICT = 'strict',
+  LAX = 'lax',
+  NONE = 'none',
+}
+
+interface ICookieStoreDeleteOptions {
+  name?: string;
+  domain?: string;
+  path?: string;
+}
+
+interface ICookieInit extends ICookieStoreDeleteOptions {
+  sameSite?: ECookieSameSite;
+}
+
+interface ICookieInitWithNameAndValue extends ICookieInit {
+  name?: string;
+  value?: string;
+}
+
+export interface IOptions extends ICookieInit {
+  defaultValue?: string;
+}
+
+interface ICookieStore {
+  get: (key: string) => Promise<ICookieInitWithNameAndValue>;
+  set: (options: ICookieInitWithNameAndValue) => Promise<void>;
+  delete: (options: ICookieStoreDeleteOptions) => Promise<void>;
+}
+
+const useCookie = (key: string, options?: IOptions) => {
+  if (!isClient) {
+    if (!isDevelopment) {
+      // eslint-disable-next-line no-console
+      return console.warn(
+        'Please be aware that cookieStore could not be available during SSR',
+      )
+    }
+
+    return Object.freeze({
+      onError: noop,
+      updateCookie: noop,
+      deleteCookie: noop,
+      cookieValue: options?.defaultValue,
+    })
+  }
+
+  if (!isAPISupported('cookieStore')) {
+    // eslint-disable-next-line no-console
+    console.warn(
+      "The current device does not support the 'cookieStore' API, you should avoid using useCookie",
+    )
+
+    return Object.freeze({
+      onError: noop,
+      updateCookie: noop,
+      deleteCookie: noop,
+      cookieValue: options?.defaultValue,
+    })
+  }
+
+  const [cookieValue, setCookieValue] = useState<string>()
+  const [onErrorRef, setOnErrorRef] = createHandlerSetter<BRHGeolocationPositionError>()
+
+  const cookieStoreObject = (window as any).cookieStore as ICookieStore
```

Comment:
why do you need to convert to `any` here?

I think Typescript already supports cookieStore out of the box, if it doesn't it's probably just a configuration issue. 
As a general role we should not re-define its interfaces ;)

Reply:
In this topic from January (https://github.com/microsoft/TypeScript/issues/47314). The guy from typescript says "
We don't include APIs that aren't in all major browsers yet" this is confirmed by looking at https://caniuse.com/?search=cookieStore.

Reply:
Oh I see... typescript strikes again! 😄

Cool, then leave it as it is...